### ISSUE TITLE
Multi master or master election

### ISSUE THREAD

**Issue by runningman84** [Graylog2/graylog2-server#810]
I would like to put graylog2 server into an autoscaling group. This won't work right now because you cannot have more than one master...

**Comment by kroepke** [comment]
Please see my comment here: https://github.com/Graylog2/graylog2-web-interface/issues/1054#issuecomment-71197324

Running two or more master nodes will lead to problems with global inputs. As such it is not recommended to run the configuration like that.

This will be fixed in a future version, but requires more thought and caution than we can afford to spent right now for 1.0-final.

**Comment by tj13** [comment]
any update for this? is there a timeline to support multi-master?

**Comment by fieryvova** [comment]
Has the situation changed?

**Comment by joschi** [comment]
@fieryvova No.

**Comment by mwysokin** [comment]
Any update on this? It is currently impossible to have a good HA infrastructure with graylog2 and hardcoded master setting. Currently we need to resort to some hacks like having an external script which changes master setting in one of the slave nodes and restarts it if the previous master goes down. 

Could any of the developers shed some light on problems with implementing this feature? Are there any blockers? Is it a problem with manpower?

**Comment by joschi** [comment]
@mwysokin If you see no progress on this issue, it's because there's no progress on this issue.

If you want to contribute a pull request tackling this problem, feel free to do so. We'll happily discuss the implementation with you.

**Comment by huegelc** [comment]
Is there a (manual) workaround for this?

**Comment by joschi** [comment]
@huegelc Yes, manually doing a failover (i. e. setting `is_master=true` on one of the secondary nodes and restart that node).

**Comment by huegelc** [comment]
@joschi This means, that on a particular point you will have 
a) no master at all
b) two master nodes
because you have to reload (old) master to become a normal node and (new) master to take over, right?

**Comment by gimmic** [comment]
Is there any update on this? I do see the problem in a big graylog cluster of running a single master(is_master=true). I also see the problem of split-brain if running multiple master-capable nodes if there's no quorum or election process.

**Comment by kroepke** [comment]
@gimmic We are working on a solution that will start being released in 3.1 and in subsequent releases we will migrate all services that currently depend on the master flag to the new system.
It might take another release or two to get rid of the flag completely though.

**Comment by razvanphp** [comment]
Hey, can we find anywhere an update on this, regarding the 3.1 version? Was checking the CHANGELOG and there is nothing about it there. Thank you!

**Comment by gimmic** [comment]
It does make me nervous there is no proper clustering / High Availability configuration for graylog. We run 30+ nodes but only have one hard-coded 'master'.

**Comment by seongpyoHong** [comment]
I tried to find an update on this issue, but I still couldn't. 
When will it be updated?

**Comment by Hmemories** [comment]
Has  this issue been solved?

**Comment by homerobono** [comment]
Interested on this feature too. A possible workaround is perform a GET `/system/cluster/node`, which returns a `is_master` parameter, or even worse, fetching with ssh like `ssh user@graylog-node "/etc/graylog/server/server.conf | grep 'is_master'"`.

This just doesn't work for me as for this to be implemented programmatically it implies in either exposing a Graylog session credentials on the code or the ssh key, for each node.

**Comment by maiconbaumx** [comment]
I don't think we will see a multi-master mode on Graylog ...
